icepyx/core/granules.py

```diff
@@ -73,16 +74,13 @@ def gran_IDs(grans, ids=False, cycles=False, tracks=False, dates=False, cloud=Fa
         gran_ids.append(producer_granule_id)
 
         if cloud is True:
-            try:
+            with contextlib.suppress(KeyError):
```

Comment:
Not a strong opinion, but I feel this is less readable despite having fewer lines.

Reply:
It just takes getting used to it is easier to read once you do.

Reply:
I think I'd need a bit more convincing on this :) Apologies, since I originally posted, my opinion has gotten a bit stronger. In my view, "taking some getting used to" is a really big cost! Code is read at least one order of magnitude more than it's written, so I care deeply about optimizing for folks who haven't gotten used to it yet.